src/UI/Component/Input/Field/Factory.php

```diff
@@ -205,4 +206,60 @@ public function dependantGroup(array $inputs);
 	 * @return    \ILIAS\UI\Component\Input\Field\Checkbox
 	 */
 	public function checkbox($label, $byline = null);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A Tag Input is used to choose a subset amount of tags (techn.: array of strings) out
+	 *     of a finite list of tags. The Tag Field SHOULD be used, whenever it is not required
+	 *     or not possible to display all available options, e.g. because the amount is too high
+	 *     when the options are "all users" or "all tags.
+	 *     Besides the tags to choose from, the user can provide own tags by typing them
+	 *     into the Input (@see TagInput::withOptionsAreExtendable ).
+	 *   composition: >
+	 *     The Input is presented as a Text Input and prepended by already selected tags
+	 *     presented as texts including a Close-Button.  (e.g. [ Amsterdam X ] )
+	 *     The input is labeled by the label given.
+	 *     Suggested tags are listed in a Dropdown-List beneath the Text Input.
+	 *   effect: >
+	 *     As soon as the user types in the Text Input, the Tag Input suggests matching tags from
+	 *     the the given list of tags. Suggestions will appear after a defined
+	 *     amount of characters, one by default.
+	 *     Clicking on one of these tags closes the List and transfers the selected tag into
+	 *     the Input, displayed as a Tag with a Close-Button.
+	 *     By clicking on a Close-Button of a already selected tag, this tag will disappear
+	 *     from the Input.
+	 *
+	 * rivals: >
+	 *     + SelectInput: Currently not part of the UI-Service.
+	 *     + Checkbox Group
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *      A Tag Input MUST NOT be used whenever a user has to perform a binary choice where
+	 *      option is automatically the inverse of the other. A Checkbox MUST be used in this case.
+	 *     2: >
+	 *      A Tag Input MUST NOT be used whenever a user has to perform a choice from a list of
+	 *      options where only one Option has to be selected. A Select MUST be used in this case
+	 *      (Not yet part of the KitchenSink).
+	 *     3: >
+	 *      A Tag Input SHOULD be used whenever a User should be able to extend the list of given options.
+	 *     4: >
+	 *      A Tag Input MUST NOT be used when a User has to choose from a finite list of options
+	 *      which can't be extended by users Input, a Multi Select MUST be used in this case
+	 *      (Not yet part of the KitchenSink).
+	 *     5: The tags provided SHOULD NOT have long titles.
+	 *
+	 * ---
+	 * @param string $label
+	 * @param string $byline
+	 * @param array  $tags    List of tags to select from, given as a list of texts
```

Comment:
We might want to use `string[]` here instead of `array` to make perfectly clear that this does not care about ids.